### PR TITLE
[improvement] Modal widget

### LINKED ISSUE #4062: Improvements to Modal widget
## Proposed Updates to the widget

- Option to trigger modal from a trigger button, javascript, and events.
- A launch/trigger button to open the modal (by default: button should be there on dropping a new modal widget)
- Add loading state to the widget
- Add Visibility (should only hide the trigger button if it's toggled on)
- Addition of two new component events: **On open** and **On close**
- Able to close the modal using the inbuilt close button in the modal header, pressing the escape key, clicking outside the modal, and with actions.
- Updates to modal sizes: medium and large
- Style Updates:  Able to change header background color, title color, and modal body color
- Style Updates for trigger button: Title and background color (when enabled).
- When the trigger button is toggled off, widget config is always displayed on the editor with a secondary background color

### PR DESCRIPTION
**Pull request by arpitnath**
Resolves #4062

### CODE REVIEW

**Comment by shubh22** [comment]
Task linked: [CU-2tzfzh8 Improvements: Modal widget](https://app.clickup.com/t/2tzfzh8)

File: frontend/src/Editor/Components/Modal.jsx [hunk 1]
```diff
@@ -35,9 +43,16 @@ export const Modal = function Modal({
     setExposedVariable('show', false);
   });
 
+  useEffect(() => {
+    setExposedVariable('show', showModal).then(() => fireEvent(showModal ? 'onOpen' : 'onClose'));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [showModal]);
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/Components/Modal.jsx:49]: `fireEvent` in the useEffect will trigger when the component is mounted

**Review by kavinvenkatachalam — CHANGES_REQUESTED** [review]
@arpitnath `onClose` event is called whenever I reload the page. It's because we're calling the `fireEvent` in the useEffect

**Review by kavinvenkatachalam — APPROVED** [review]
LGTM

**Comment by sherfin94** [comment]
@arpitnath Please test the updated migration once

**Review by kavinvenkatachalam — APPROVED** [review]

**Pull request merged**